### PR TITLE
Resolve timeout handling regression in 10.0.5 in new Transports style

### PR DESCRIPTION
**Pull request by ktoso**
Resolves https://github.com/akka/akka-http/issues/1012

Though we may want to discuss the entire settings design in depth rather soon.
See also: https://github.com/akka/akka-http/issues/1009

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/javadsl/ClientTransport.scala [hunk 1]
```diff
@@ -39,15 +40,15 @@ object ClientTransport {
...
-      JavaMapping.toJava(delegate.connectTo(host, port)(system))
+      JavaMapping.toJava(delegate.connectTo(host, port, settings)(system))
     }
   }
   private class JavaWrapper(val delegate: ClientTransport) extends scaladsl.ClientTransport {
-    def connectTo(host: String, port: Int)(implicit system: ActorSystem): akka.stream.scaladsl.Flow[ByteString, ByteString, Future[scaladsl.Http.OutgoingConnection]] = {
+    def connectTo(host: String, port: Int, settings: ClientConnectionSettings)(implicit system: ActorSystem): akka.stream.scaladsl.Flow[ByteString, ByteString, Future[scaladsl.Http.OutgoingConnection]] = {
       import system.dispatcher
-      JavaMapping.toScala(delegate.connectTo(host, port, system))
+      ??? // JavaMapping.toScala(delegate.connectTo(host, port, system))
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/javadsl/ClientTransport.scala]: fixme

**Comment by jrudolph** [comment]
It's not clear to me (as it didn't seem to be to the original reporter) that something is broken here and what that should be. Can you explain?

This PR reverts the ClientTransport feature which I guess was unintended? Or was that the resolution?

It seems to contain some nice refactorings that could make sense in another PR.

**Comment by ktoso** [comment]
Let's it around and let's discuss it next week - AFAIR the situation is that the setting was taken from the "initial", place and whatever people would be setting when calling outgoingConnection (by providing the settings in there) ends up being ignored with the current transports version.

Could be that my impl here is still not completely correct - there's a lot of dancing around the settings / passing and what it should actually mean in general..

**Comment by ktoso** [comment]
@jrudolph could I ask you for a review picking the things/refactorings you'd like from this PR?

**Comment by jrudolph** [comment]
I think I'll follow up on this one. It's me who broke it with the `ClientTransport`, so I'll have another look about how to fix it and I'll take the ideas from this one if possible.

**Comment by ktoso** [comment]
Yeah go ahead and take it if you want :) Shall we close this one?

**Comment by ktoso** [comment]
Just FYI: This is tagged as pick next, but in no shape to get into today's release.
Let me know if I should close it (afair you wanted to take it over @jrudolph)

**Comment by jrudolph** [comment]
Superseded by #1195.

**Comment by ktoso** [comment]
Great, thanks. Will review :)

**Pull request closed without merging**